### ISSUE TITLE
Uncaught TypeError: Path must be a string. Received null

### ISSUE THREAD

**Issue by plaxdan** [mukeshsoni/elm-instant#9]
[Enter steps to reproduce below:]
1. `Cmd+Shift+P`
2. Type `elm inst` and hit enter on the correct entry
3. 💥 

**Atom Version**: 1.10.2
**Electron Version**: 0.37.8
**System**: Mac OS X 10.12.1
**Thrown From**: [elm-instant](https://github.com/mukeshsoni/elm-instant) package, v0.4.0
### Stack Trace

Uncaught TypeError: Path must be a string. Received null

```
At path.js:7

TypeError: Path must be a string. Received null
    at assertPath (path.js:7:11)
    at Object.resolve (path.js:1188:7)
    at writeSourcesToElmPackageJson (/Users/daniel/.atom/packages/elm-instant/lib/compilers/elm-compiler/index.js:78:89)
    at updateFileSources (/Users/daniel/.atom/packages/elm-instant/lib/compilers/elm-compiler/index.js:123:10)
    at Object.compile (/Users/daniel/.atom/packages/elm-instant/lib/compilers/elm-compiler/index.js:357:10)
    at Object.onCodeChange (/Users/daniel/.atom/packages/elm-instant/lib/compilers/elm-compiler/index.js:455:15)
    at Object.compile (/Users/daniel/.atom/packages/elm-instant/lib/elm-instant.js:158:16)
    at Object.playgroundCodeChanged (/Users/daniel/.atom/packages/elm-instant/lib/elm-instant.js:145:10)
    at Function.module.exports.Emitter.simpleDispatch (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/text-buffer/node_modules/event-kit/lib/emitter.js:25:14)
    at Emitter.module.exports.Emitter.emit (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/text-buffer/node_modules/event-kit/lib/emitter.js:129:28)
    at /Applications/Atom.app/Contents/Resources/app.asar/node_modules/text-buffer/lib/text-buffer.js:1429:25
```
### Commands

```
     -0:02.1.0 command-palette:toggle (div.panels)
     -0:01.9.0 blur (div.panels)
     -0:00.6.0 core:confirm (atom-text-editor.editor.mini.is-focused)
     -0:00.6.0 blur (atom-text-editor.editor.mini.is-focused)
     -0:00.6.0 elm-instant:toggle (div.panels)
  3x -0:00.5.0 blur (div.panels)
```
### Config

``` json
{
  "core": {
    "customFileTypes": {
      "source.ini": [
        ".buckconfig",
        ".flowconfig"
      ],
      "source.json": [
        ".arcconfig"
      ],
      "source.python": [
        "BUCK"
      ]
    },
    "disabledPackages": [
      "pretty-json"
    ],
    "packagesWithKeymapsDisabled": [
      "markdown-preview",
      "highlight-selected"
    ],
    "themes": [
      "one-light-ui",
      "one-light-syntax"
    ]
  }
}
```
### Installed Packages

``` coffee
# User
advanced-open-file, v0.16.4 (inactive)
atom-alignment, v0.12.1 (active)
atom-beautify, v0.29.13 (inactive)
atom-live-server, v1.2.5 (inactive)
atom-yeoman, v0.3.15 (active)
autocomplete-paths, v1.0.2 (active)
coffee-compile, v0.22.0 (active)
duotone-blue-dark-syntax, v1.4.0 (inactive)
elm-format, v2.1.0 (active)
elm-instant, v0.4.0 (active)
file-icons, v1.7.19 (active)
filesize, v0.4.2 (active)
git-blame, v0.4.11 (inactive)
git-plus, v5.18.0 (active)
git-time-machine, v1.5.3 (inactive)
goto, v1.7.1 (inactive)
highlight-selected, v0.11.2 (active)
ionide-fake, v1.2.2 (active)
ionide-fsharp, v1.9.1 (active)
ionide-fsi, v2.1.3 (active)
ionide-installer, v1.3.0 (active)
ionide-paket, v2.2.6 (active)
ionide-webview, v1.0.3 (active)
ionide-yeoman, v1.0.2 (active)
language-elm, v1.5.0 (active)
language-fsharp, v0.8.4 (active)
linter, v1.11.16 (active)
linter-elm-make, v0.17.7 (active)
linter-jsonlint, v1.3.0 (active)
lumens-syntax, v0.1.0 (inactive)
markdown-scroll-sync, v2.1.2 (active)
minimap, v4.25.0 (active)
minimap-cursorline, v0.2.0 (active)
minimap-find-and-replace, v4.5.1 (active)
minimap-git-diff, v4.3.1 (active)
minimap-highlight-selected, v4.4.0 (active)
minimap-selection, v4.4.0 (active)
monokai-light, v0.3.7 (inactive)
open-terminal-here, v2.1.4 (inactive)
paradise-light-syntax, v0.3.0 (inactive)
pigments, v0.36.0 (active)
set-syntax, v0.3.0 (active)
sort-lines, v0.14.0 (inactive)
sync-settings, v0.7.2 (active)
todo-show, v1.7.0 (inactive)
atom-dark-syntax, v0.27.0 (inactive)
atom-dark-ui, v0.52.0 (inactive)
atom-light-syntax, v0.28.0 (inactive)
atom-light-ui, v0.44.0 (inactive)
base16-tomorrow-dark-theme, v1.1.0 (inactive)
base16-tomorrow-light-theme, v1.1.1 (inactive)
one-dark-ui, v1.5.0 (inactive)
one-light-ui, v1.5.0 (active)
one-dark-syntax, v1.3.0 (inactive)
one-light-syntax, v1.3.0 (active)
solarized-dark-syntax, v1.0.2 (inactive)
solarized-light-syntax, v1.0.2 (inactive)
about, v1.5.3 (active)
archive-view, v0.61.1 (active)
autocomplete-atom-api, v0.10.0 (active)
autocomplete-css, v0.11.2 (active)
autocomplete-html, v0.7.2 (active)
autocomplete-plus, v2.31.1 (active)
autocomplete-snippets, v1.11.0 (active)
autoflow, v0.27.0 (inactive)
autosave, v0.23.1 (active)
background-tips, v0.26.1 (active)
bookmarks, v0.41.0 (active)
bracket-matcher, v0.82.1 (active)
command-palette, v0.38.0 (active)
deprecation-cop, v0.54.1 (active)
dev-live-reload, v0.47.0 (active)
encoding-selector, v0.22.0 (active)
exception-reporting, v0.39.0 (active)
find-and-replace, v0.201.0 (inactive)
fuzzy-finder, v1.3.0 (active)
git-diff, v1.1.0 (active)
go-to-line, v0.31.0 (inactive)
grammar-selector, v0.48.1 (active)
image-view, v0.58.2 (active)
incompatible-packages, v0.26.1 (active)
keybinding-resolver, v0.35.0 (active)
line-ending-selector, v0.5.0 (active)
link, v0.31.1 (inactive)
markdown-preview, v0.158.0 (active)
metrics, v0.53.1 (active)
notifications, v0.65.0 (active)
open-on-github, v1.2.0 (inactive)
package-generator, v1.0.0 (inactive)
settings-view, v0.242.2 (active)
snippets, v1.0.2 (active)
spell-check, v0.67.1 (active)
status-bar, v1.4.1 (active)
styleguide, v0.47.0 (active)
symbols-view, v0.113.0 (inactive)
tabs, v0.100.2 (active)
timecop, v0.33.2 (active)
tree-view, v0.208.2 (active)
update-package-dependencies, v0.10.0 (active)
welcome, v0.34.0 (active)
whitespace, v0.33.0 (active)
wrap-guide, v0.38.1 (active)
language-c, v0.52.1 (active)
language-clojure, v0.21.0 (active)
language-coffee-script, v0.47.2 (active)
language-csharp, v0.12.1 (active)
language-css, v0.37.1 (active)
language-gfm, v0.88.0 (active)
language-git, v0.15.0 (active)
language-go, v0.42.1 (active)
language-html, v0.45.1 (active)
language-hyperlink, v0.16.0 (active)
language-java, v0.23.0 (active)
language-javascript, v0.119.0 (active)
language-json, v0.18.2 (active)
language-less, v0.29.5 (active)
language-make, v0.22.2 (active)
language-mustache, v0.13.0 (active)
language-objective-c, v0.15.1 (active)
language-perl, v0.35.0 (active)
language-php, v0.37.2 (active)
language-property-list, v0.8.0 (active)
language-python, v0.45.0 (active)
language-ruby, v0.69.0 (active)
language-ruby-on-rails, v0.25.0 (active)
language-sass, v0.56.0 (active)
language-shellscript, v0.22.4 (active)
language-source, v0.9.0 (active)
language-sql, v0.23.0 (active)
language-text, v0.7.1 (active)
language-todo, v0.28.0 (active)
language-toml, v0.18.0 (active)
language-xml, v0.34.9 (active)
language-yaml, v0.26.0 (active)

# Dev
No dev packages
```

**Comment by mukeshsoni** [comment]
Hi @plaxdan, thanks for reporting the issue. Do you have a file open when you toggle elm-instant?

**Comment by plaxdan** [comment]
In this case I had no files open. After viewing [this gif](https://raw.githubusercontent.com/mukeshsoni/elm-instant/master/images/showcase.gif) my expectation was that triggering the elm-instant command would open two tabs:
-  a new (untitled) file and that I could start typing in
- and the preview tab which would show code/compiler output

**Comment by mukeshsoni** [comment]
Thanks @plaxdan! Have pushed a fix in 0.4.2. Upgrade the package in atom and let me know if it still persists.

**Comment by plaxdan** [comment]
Upgraded and confirmed that no error is thrown. Thank you!
